src/builtin.d

```diff
@@ -48,35 +49,35 @@ extern (C++) Expression eval_sin(Loc loc, FuncDeclaration fd, Expressions* argum
 {
     Expression arg0 = (*arguments)[0];
     assert(arg0.op == TOKfloat64);
-    return new RealExp(loc, sinl(arg0.toReal()), arg0.type);
+    return new RealExp(loc, TargetReal.sin(arg0.toReal()), arg0.type);
```

Comment:
Again, I'm not sure what these math functions have to do with the layout/properties of ``real_t``.  I really do think they should all be part of ``Port``, which is already used for doing things on the host in an agnostic manner (even though DMD may not follow that precise specification, it's how GDC treats it).